src/ai/backend/common/redis_helper.py

```diff
@@ -485,34 +447,44 @@ def get_redis_object(
         else:
             sentinel_addresses = _sentinel_addresses
 
-        assert redis_config.get("service_name") is not None
+        service_name = redis_config.get("service_name")
+        password = redis_config.get("password")
+        assert (
+            service_name is not None
+        ), "config/redis/service_name is required when using Redis Sentinel"
+
         sentinel = Sentinel(
             [(str(host), port) for host, port in sentinel_addresses],
-            password=redis_config.get("password"),
+            password=password,
             db=str(db),
             sentinel_kwargs={
+                "password": password,
                 **kwargs,
             },
         )
+
         return RedisConnectionInfo(
-            client=sentinel,
-            service_name=redis_config.get("service_name"),
+            client=sentinel.master_for(service_name=service_name, **_default_conn_opts),
```

Comment:
I think we should preserve `socket_connect_timeout=reconnect_poll_interval` kwarg here.